src/System.Text.Json/tests/JsonStringTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonStringTests
+    {
+        [Fact]
+        public static void TestDefaultCtor()
+        {
+            var jsonString = new JsonString();
+            Assert.Equal("", jsonString.Value);
+        }
+
+        [InlineData("value")]
+        [InlineData("value with some spaces")]
+        [InlineData("     leading spaces")]
+        [InlineData("trailing spaces"     )]
+        [InlineData("new lines\r\n")]
+        [InlineData("tabs\ttabs\t")]
+        [InlineData("\\u003e\\u003e\\u003e\\u003e\\u003e")]
+        [InlineData("zażółć gęślą jaźń")]
+        [InlineData("\u6f22\u5b57 \u6f22\u5b57")]
+        [InlineData(">><++>>>\">>\\>>&>>>\u6f22\u5B57>>>")]
+        [InlineData("Here is a string: \\\"\\\"\":\"Here is a\",\"Here is a back slash\\\\\":[\"Multiline\\r\\n String\\r\\n\",\"\\tMul\\r\\ntiline String\",\"\\\"somequote\\\"\\tMu\\\"\\\"l\\r\\ntiline\\\"another\\\" String\\\\\"],\"str")]
+        [InlineData("Hello / a / b / c \\/ \\r\\b\\n\\f\\t\\/")]
+        [Theory]
+        public static void TestInitialization(string value)
+        {
+            // Default constructor:
+            var defaultlyInitializedJsonString = new JsonString();
+            defaultlyInitializedJsonString.Value = value;
+            Assert.Equal(value, defaultlyInitializedJsonString.Value);
+
+            // To string:
+            Assert.Equal(value, defaultlyInitializedJsonString.ToString());
+
+            // Value constructor:
+            Assert.Equal(value, new JsonString(value).Value);
+
+            // Implicit operator:            
+            JsonString implicitlyInitializiedJsonString = value;
+            Assert.Equal(value, implicitlyInitializiedJsonString.Value);
+
+            // Casted to span:            
+#if BUILDING_INBOX_LIBRARY
+            ReadOnlySpan<char> spanValue = value;
+            Assert.Equal(value, new JsonString(spanValue).Value);
+#endif
+        }
+
+        [Fact]
+        public static void TestNulls()
+        {
+#if BUILDING_INBOX_LIBRARY
+            Assert.Throws<ArgumentNullException>(() => new JsonString(null));
+#endif
+            Assert.Throws<ArgumentNullException>(() => new JsonString().Value = null);
+        }
+
+#if !BUILDING_INBOX_LIBRARY
```

Comment:
```suggestion
#if BUILDING_INBOX_LIBRARY
```
Should this be flipped from not inbox to inbox? Its only running the test on `Windows NETFX_x86_Release` which is where its failing?

Reply:
I deleted platform checks. I observed it throwing exception on netfx locally, but I was running tests incorrectly. The differences in behavior on netfx and corefx were due to lack of implicit cast from string to ReadOnlySpan in netfx.